jerry-core/lit/lit-literal.cpp

```diff
@@ -470,39 +492,38 @@ lit_literal_equal_type (literal_t lit1, /**< first literal */
  * @return pointer to the zero-terminated string.
  */
 const lit_utf8_byte_t *
-lit_literal_to_utf8_string (literal_t lit, /**< literal to be processed */
+lit_literal_to_utf8_string (lit_literal_t lit, /**< literal to be processed */
                             lit_utf8_byte_t *buff_p, /**< buffer to use as a string storage */
                             size_t size) /**< size of the buffer */
 {
   JERRY_ASSERT (buff_p != NULL && size > 0);
-  rcs_record_t::type_t type = lit->get_type ();
+  rcs_record_type_t type = rcs_record_get_type (lit);
 
-  switch (type)
+  if (RCS_RECORD_TYPE_IS_CHARSET (type))
   {
-    case LIT_STR_T:
-    {
-      lit_charset_record_t *ch_rec_p = static_cast<lit_charset_record_t *> (lit);
-      ch_rec_p->get_charset (buff_p, size);
-      return buff_p;
-    }
-    case LIT_MAGIC_STR_T:
-    {
-      return lit_get_magic_string_utf8 (lit_magic_record_get_magic_str_id (lit));
-    }
-    case LIT_MAGIC_STR_EX_T:
-    {
-      return lit_get_magic_string_ex_utf8 (lit_magic_record_ex_get_magic_str_id (lit));
-    }
-    case LIT_NUMBER_T:
-    {
-      ecma_number_to_utf8_string (static_cast<lit_number_record_t *> (lit)->get_number (), buff_p, (ssize_t)size);
+    rcs_record_get_charset (&rcs_lit_storage, lit, buff_p, size);
+    return buff_p;
+  }
 
-      return buff_p;
-    }
-    default: JERRY_UNREACHABLE ();
+  if (RCS_RECORD_TYPE_IS_MAGIC_STR (type))
+  {
+    return lit_get_magic_string_utf8 (rcs_record_get_magic_str_id (lit));
+  }
+
+  if (RCS_RECORD_TYPE_IS_MAGIC_STR_EX (type))
+  {
+    return lit_get_magic_string_ex_utf8 (rcs_record_get_magic_str_ex_id (lit));
+  }
+
+  if (RCS_RECORD_TYPE_IS_NUMBER (type))
+  {
+    ecma_number_t number = rcs_record_get_number (&rcs_lit_storage, lit);
+    ecma_number_to_utf8_string (number, buff_p, (ssize_t) size);
+    return buff_p;
   }
 
   JERRY_UNREACHABLE ();
```

Comment:
As `JERRY_UNREACHABLE` is not removed in release build, its better to avoid including it here.
Instead, its better to convert the above sequence of `if` statements to `if` / `else if` / `else if` / `else` with assertion `RCS_RECORD_TYPE_IS_NUMBER (type)` in the `else` block.